tests/providers/amazon/aws/operators/test_redshift_data.py

```diff
@@ -140,3 +140,39 @@ def test_batch_execute(self, mock_conn):
             StatementName=statement_name,
             WithEvent=False,
         )
+
+    @mock.patch("airflow.providers.amazon.aws.hooks.redshift_data.RedshiftDataHook.conn")
+    def test_return_sql_result(self, mock_conn):
+        expected_result = {"Result": True}
+        mock_conn.execute_statement.return_value = {"Id": STATEMENT_ID}
```

Comment:
```suggestion
        mock_conn.batch_execute_statement.return_value = {"Id": STATEMENT_ID}
```

You need to mock `batch_execute_statement` here because you call

https://github.com/apache/airflow/blob/2a34dc9e8470285b0ed2db71109ef4265e29688b/airflow/providers/amazon/aws/operators/redshift_data.py#L124-L136

You could check it locally in Breeze or venv, for more detail see: [Airflow Unit Tests](https://github.com/apache/airflow/blob/main/TESTING.rst#id3)

Reply:
Fixed. Actually, I needed `execute_statement`, not `batch_execute_statement`. Looks like it was a copy-paste problem.